packages/legend-application-studio/src/stores/editor/sidebar-state/BulkServiceRegistrationState.ts

```diff
@@ -48,54 +82,159 @@ export class BulkServiceRegistrationState extends ServiceConfigState {
     );
 
     makeObservable(this, {
-      showSuccessModel: observable,
+      editorStore: false,
+      service: observable,
+      registrationResult: observable,
+      setShowFailingView: action,
+      setSelectService: action,
+      isSelected: observable,
+    });
+    this.globalBulkServiceRegistrationState =
+      globalBulkServiceRegistrationState;
+    this.service = service;
+  }
+
+  setSelectService(): void {
+    this.isSelected ? (this.isSelected = false) : (this.isSelected = true);
+  }
+  setShowFailingView(val: boolean): void {
+    this.showFailingView = val;
+  }
+
+  handleRegistrationResult(
+    registrationResult: ServiceRegistrationResult,
+  ): void {
+    try {
+      assertTrue(registrationResult.service?.pattern === this.service.pattern);
+      this.registrationResult = registrationResult;
+    } catch (error) {
+      assertErrorThrown(error);
+    }
+  }
+}
+
+export class GlobalBulkServiceRegistrationState {
+  editorStore: EditorStore;
+  sdlcState: EditorSDLCState;
+  bulkServiceRegistrationState: BulkServiceRegistrationState[] | undefined;
+  serviceConfigState: ServiceConfigState;
+  isServiceRegistering = ActionState.create();
+  showRegistrationConfig = false;
+  failingView: ServiceRegistrationFail | undefined;
+  selectAllServices = false;
+  numOfServicesSelected = 0;
```

Comment:
dont see the need to have this on the state or to calculate. 
can't we just render this in the editor
````
this.bulkServiceRegistrationState?.map(e => e.isSelected).length
```

Reply:
got rid of variables and rendered with the this: 

`this.bulkServiceRegistrationState?.map(e => e.isSelected).length`